File: Python/civitasacademic.py
from human import Human # Import class Human dari file human.py

class CivitasAcademic(Human): # Deklarasi class CivitasAcademic (child class dari Human)
    __university = "" # Untuk menyimpan nama universitas, atribut berjenis private
    __email = "" # Untuk menyimpan email mahasiswa, atribut berjenis private
    __faculty = "" # Untuk menyimpan fakultas mahasiswa, atribut berjenis private

    def __init__(self, identity = 0, name = "", gender = '', university = "", email = "", faculty = ""): # Membuat konstruktor
        super().__init__(identity, name, gender) # Menginisialisasi setiap atribut yang ada di dalam konstruktor parent class Human
        self.__university = university # Menginisialisasi atribut __university
        self.__email = email # Menginisialisasi atribut __email
        self.__faculty = faculty # Menginisialisasi atribut __faculty

    def setUniversity(self, university): # Setter untuk atribut __university
        self.__university = university # Menginisialisasi atribut __university

    def setEmail(self, email): # Setter untuk atribut __email
        self.__email = email # Menginisialisasi atribut __email

    def setFaculty(self, faculty): # Setter untuk atribut __faculty
        self.__faculty = faculty # Menginisialisasi atribut __faculty

    def getUniversity(self): # Getter untuk atribut __university
        return self.__university # Mengembalikan nilai atribut __university

    def getEmail(self): # Getter untuk atribut __email
        return self.__email # Mengembalikan nilai atribut __email
    
    def getFaculty(self): # Getter untuk atribut __faculty
        return self.__faculty # Mengembalikan nilai atribut __faculty

File: C++/CivitasAcademic.cpp
// Import library yang dibutuhkan
#include <bits/stdc++.h>

using namespace std; // Menggunakan standard namespace

class CivitasAcademic : public Human // Membuat class CivitasAcademic (child) yang mengextends ke class Human (parent)
{
private:               // Membuat atribut berjenis private
    string university; // Untuk menyimpan nama universitas mahasiswa
    string faculty;    // Untuk menyimpan nama fakultas mahasiswa
    string email;      // Untuk menyimpan email mahasiswa

public:
    /* Konstruktor */
    CivitasAcademic() : Human() // Membuat konstruktor tanpa isian, class yang di-extends akan diinstansiasi terlebih dahulu
    {
        this->university = "";
        this->faculty = "";
        this->email = "";
    }

    CivitasAcademic(long long identity, string name, char gender, string university, string faculty, string email) : Human(identity, name, gender) // Membuat konstruktor dengan isian dari parameter
    {
        this->university = university;
        this->faculty = faculty;
        this->email = email;
    }

    /* Setter dan Getter untuk setiap atribut di dalam class CivitasAcademic */
    void setUniversity(string university)
    {
        this->university = university;
    }

    void setFaculty(string faculty)
    {
        this->faculty = faculty;
    }

    void setEmail(string email)
    {
        this->email = email;
    }

    string getUniversity()
    {
        return this->university;
    }

    string getFaculty()
    {
        return this->faculty;
    }

    string getEmail()
    {
        return this->email;
    }

    /* Destruktor */
    ~CivitasAcademic()
    {
    }
};